lntest/itest/lnd_channel_force_close.go

```diff
@@ -0,0 +1,223 @@
+		defer shutdownAndAssert(net, t, alice)
+
+		bob := setupNode("Bob")
+		defer shutdownAndAssert(net, t, bob)
+
+		// Connect Alice to Bob.
+		err := net.ConnectNodes(ctxt, alice, bob)
+		require.NoError(t.t, err, "unable to connect alice to bob")
+
+		// Open a channel between Alice and Bob.
+		chanPoint := openChannelAndAssert(
+			ctxt, t, net, alice, bob,
+			lntest.OpenChannelParams{
+				Amt:     10e6,
+				PushAmt: 5e6,
+			},
+		)
+
+		// Send a payment with a specified finalCTLVDelta, which will
+		// be used as our deadline later on when Alice force closes the
+		// channel.
+		_, err = alice.RouterClient.SendPaymentV2(
+			ctxt,
+			&routerrpc.SendPaymentRequest{
+				Dest:           bob.PubKey[:],
+				Amt:            10e4,
+				PaymentHash:    makeFakePayHash(t),
+				FinalCltvDelta: finalCTLV,
+				TimeoutSeconds: 60,
+				FeeLimitMsat:   noFeeLimitMsat,
+			},
+		)
+		require.NoError(t.t, err, "unable to send alice htlc")
+
+		// Once the HTLC has cleared, all the nodes in our mini network
+		// should show that the HTLC has been locked in.
+		nodes := []*lntest.HarnessNode{alice, bob}
+		err = wait.NoError(func() error {
+			return assertNumActiveHtlcs(nodes, 1)
+		}, defaultTimeout)
+		require.NoError(t.t, err, "htlc mismatch")
+
+		// Alice force closes the channel.
+		_, _, err = net.CloseChannel(ctxt, alice, chanPoint, true)
+		require.NoError(t.t, err, "unable to force close channel")
+
+		// Now that the channel has been force closed, it should show
+		// up in the PendingChannels RPC under the waiting close
+		// section.
+		pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+		pendingChanResp, err := alice.PendingChannels(
+			ctxt, pendingChansRequest,
+		)
+		require.NoError(
+			t.t, err, "unable to query for pending channels",
+		)
+		require.NoError(
+			t.t, checkNumWaitingCloseChannels(pendingChanResp, 1),
+		)
+
+		// We should see only one sweep transaction because the anchor
+		// sweep is skipped.
+		sweepTxns, err := getNTxsFromMempool(
+			net.Miner.Client,
+			expectedSweepTxNum, minerMempoolTimeout,
+		)
+		require.NoError(
+			t.t, err, "failed to find commitment tx in mempool",
+		)
+
+		// Mine a block to confirm these transactions such that they
+		// don't remain in the mempool for any subsequent tests.
+		_, err = net.Miner.Client.Generate(1)
+		require.NoError(t.t, err, "unable to mine blocks")
+
+		// Calculate the fee rate used.
+		feeRate := calculateTxnsFeeRate(t.t, net.Miner, sweepTxns)
+
+		return feeRate
+	}
+
+	// Setup our fee estimation for the deadline. Because the fee rate is
+	// smaller than the parent tx's fee rate, this value won't be used and
+	// we should see only one sweep tx in the mempool.
+	net.SetFeeEstimateWithConf(feeRateSmall, deadline)
+
+	// Calculate fee rate used.
+	feeRate := calculateSweepFeeRate(1)
+
+	// We expect the default max fee rate is used. Allow some deviation
+	// because weight estimates during tx generation are estimates.
+	require.InEpsilonf(
```

Comment:
Nice test function, at times stuff like fees and be pretty fuzzy so I far prefer this to having some brittle arithmetic inline within the test itself.